### ISSUE TITLE
[Bug]: ESM and AMD folders are missing in the 8.108.3 update

### ISSUE THREAD

**Issue by waelbettayeb** [microsoft/fluentui#27732]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 142.19 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 112.0.5615.137
    Chrome Canary: 115.0.5743.0
    Edge: 112.0.1722.64
    Firefox: 112.0.2
    Safari: 16.4
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://aka.ms/fluentpen

## Actual Behavior
The lib (esm) and lib-amd folders are missing in the v8.108.3, this is causing a dependency bug for all version 8 lib.
https://www.npmjs.com/package/@fluentui/react/v/8.108.3?activeTab=code

## Expected Behavior
These folders should present like:
https://www.npmjs.com/package/@fluentui/react/v/8.108.2?activeTab=code

### Requested priority

High

### Products/sites affected

fluentui/react^8

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by JustSlone** [comment]
Assigning to @bsunderhus since he's on support. But anyone from @microsoft/fluentui-react-build can feel free to pick this up.

**Comment by waelbettayeb** [comment]
> Hmm, that's not good. We'll look into this soon.
> 
> In the meantime 8.108.2 has the necessary folders, so if you are blocked, dropping down to this version should get you unblocked

Dropping down to this version won't fix the problem because it has also dependencies to package that has the same issue.

**Comment by pensphreak** [comment]
Yeah, for example @fluent/date-time-utilities 8.5.8 has the same problem

[](https://www.npmjs.com/package/@fluentui/date-time-utilities/v/8.5.8)

**Comment by jake4e** [comment]
Similar issue after downgrading as well.
Error: Could not load /node_modules/@fluentui/react-hooks/node_modules/@fluentui/utilities/lib/index.js (imported by node_modules/@fluentui/react-hooks/lib/index.js): ENOENT: no such file or directory, open '/node_modules/@fluentui/react-hooks/node_modules/@fluentui/utilities/lib/index.js'

**Comment by rudymiked** [comment]
any solutions? I am blocked by this as well. I attempted to create a bug but was blocked.

**Comment by JustSlone** [comment]
It doesn't make sense that the old versions are failing. Are you sure the lockfile isn't picking up the new version instead of the older one? You might need to pin the version and regenerate it, or even use the `resolutions` field to force a resolution.

**Comment by TristanWatanabe** [comment]
These are the version numbers of the last known good release of `@fluentui/react` and all its fluent dependencies. While we try to investigate what caused the issue and release a new version, we recommend pinning your fluent dependencies to the versions below for the time being.
```	
	"@fluentui/react": "8.108.2",
    "@fluentui/date-time-utilities": "8.5.7",
    "@fluentui/foundation-legacy": "8.2.33",
    "@fluentui/font-icons-mdl2": "8.5.13",
    "@fluentui/merge-styles": "8.5.7",
    "@fluentui/react-focus": "8.8.19",
    "@fluentui/react-hooks": "8.6.20",
    "@fluentui/react-portal-compat-context": "9.0.5",
    "@fluentui/react-window-provider": "2.2.9",
    "@fluentui/set-version": "8.2.6",
    "@fluentui/style-utilities": "8.9.6",
    "@fluentui/theme": "2.6.25",
    "@fluentui/utilities": "8.13.9"
```

**Comment by rudymiked** [comment]
![image](https://user-images.githubusercontent.com/6699164/235545438-be7379b2-d53f-4687-acc7-e4999ef547ee.png)
What I am seeing after Tristan's suggestion

**Comment by JustSlone** [comment]
> ![image](https://user-images.githubusercontent.com/6699164/235545438-be7379b2-d53f-4687-acc7-e4999ef547ee.png) What I am seeing after Tristan's suggestion

Can you confirm which version of the package is being used? 

If it's still the old one you probably need to pin the version:
- https://classic.yarnpkg.com/en/docs/selective-version-resolutions/
- https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

and you'll probably need `yarn --force` or `npm install -f` to force it to update

**Comment by rudymiked** [comment]
![image](https://user-images.githubusercontent.com/6699164/235546490-ab6bd601-0c21-4167-b634-9958972d4bd6.png)

**Comment by rudymiked** [comment]
I do see a concerning dependency in my package-lock.json:


![image](https://user-images.githubusercontent.com/6699164/235546815-8efd97e0-9f36-483e-8b1e-b4dedf928f8d.png)

**Comment by JustSlone** [comment]
Yeah, you’ll need to use overrides/resolutions to force the package downgrade

this is likely due to inter package dependencies using ^ (so even if the mail package is downgraded the dependencies grave the latest) which can be good sometimes, but not now.

**Comment by shaominFei** [comment]
I have the same issues as rudymiked's after Tristan's suggestion. Here is my package.json
![image](https://user-images.githubusercontent.com/90332261/235550505-276053f3-f8a8-46a0-9fb5-e2aee03a073c.png)

and here is the package version in my local computer.

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
└── **@fluentui/react@8.108.2**

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls fluentui/react-focus     
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
└── (empty)

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react-focus    
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├── **@fluentui/react-focus@8.8.19**
└─┬ @fluentui/react@8.108.2
  └── @fluentui/react-focus@8.8.19 deduped

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react-hooks
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├── **@fluentui/react-hooks@8.6.20**
└─┬ @fluentui/react@8.108.2
  └── @fluentui/react-hooks@8.6.20 deduped

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react-icons
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
└── **@fluentui/react-icons@2.0.201**

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react-portal-compat-context
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├── **@fluentui/react-portal-compat-context@9.0.5**
└─┬ @fluentui/react@8.108.2
  └── @fluentui/react-portal-compat-context@9.0.5 deduped

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/react-window-provider
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├─┬ **@fluentui/react-hooks@8.6.20**
│ └── @fluentui/react-window-provider@2.2.9 deduped
├── @fluentui/react-window-provider@2.2.9
└─┬ @fluentui/react@8.108.2
  └── @fluentui/react-window-provider@2.2.9 deduped

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/set-version    
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├─┬ **@fluentui/date-time-utilities@8.5.7**
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/font-icons-mdl2@8.5.13
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/foundation-legacy@8.2.33
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/merge-styles@8.5.7
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/react-focus@8.8.19
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/react-hooks@8.6.20
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/react-window-provider@2.2.9
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/react@8.108.2
│ └── @fluentui/set-version@8.2.6 deduped
├── @fluentui/set-version@8.2.6
├─┬ @fluentui/style-utilities@8.9.6
│ └── @fluentui/set-version@8.2.6 deduped
├─┬ @fluentui/theme@2.6.25
│ └── @fluentui/set-version@8.2.6 deduped
└─┬ @fluentui/utilities@8.13.9
  ├─┬ @fluentui/dom-utilities@2.2.7
  │ └── @fluentui/set-version@8.2.7
  └── @fluentui/set-version@8.2.6 deduped

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web>




                                                                                                       npm ls @fluentui/style-utilities

build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├─┬ **@fluentui/font-icons-mdl2@8.5.13**
│ └── @fluentui/style-utilities@8.9.6 deduped
├─┬ @fluentui/foundation-legacy@8.2.33
│ └── @fluentui/style-utilities@8.9.6 deduped
├─┬ @fluentui/react-focus@8.8.19
│ └── @fluentui/style-utilities@8.9.6 deduped
├─┬ @fluentui/react@8.108.2
│ └── @fluentui/style-utilities@8.9.6 deduped
└── @fluentui/style-utilities@8.9.6

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/theme          
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├─┬ **@fluentui/react@8.108.2**
│ └── @fluentui/theme@2.6.25 deduped
├─┬ @fluentui/style-utilities@8.9.6
│ └── @fluentui/theme@2.6.25 deduped
└── @fluentui/theme@2.6.25

PS C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web> npm ls @fluentui/utilities      
build-deploy@0.1.0 C:\Users\sfei\OneDrive - Online Business Systems\Desktop\work\HumanWise\projects\Web\HumanWise.Web
├─┬ **@fluentui/font-icons-mdl2@8.5.13**
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/foundation-legacy@8.2.33
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/react-focus@8.8.19
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/react-hooks@8.6.20
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/react@8.108.2
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/style-utilities@8.9.6
│ └── @fluentui/utilities@8.13.9 deduped
├─┬ @fluentui/theme@2.6.25
│ └── @fluentui/utilities@8.13.9 deduped
└── @fluentui/utilities@8.13.9

**Comment by TristanWatanabe** [comment]
`@fluentui/react` latest version [8.109.0](https://www.npmjs.com/package/@fluentui/react?activeTab=code) is now released which fixes the missing esm and amd folder issues

**Comment by rudymiked** [comment]
I am still seeing this error: 
![image](https://user-images.githubusercontent.com/6699164/235730950-30849788-ce43-4afb-b908-e3091b2bcfc4.png)

I added scheme-utilities and see the same error:

![image](https://user-images.githubusercontent.com/6699164/235731019-eb53e54f-b80c-4d19-8936-df20cddfc3ec.png)

**Comment by sak797** [comment]
There seems to be some issue resolving @fluentui/react-icons-mdl2
[commonjs--resolver] Failed to resolve entry for package "@fluentui/react-icons-mdl2". The package may have incorrect main/module/exports specified in its package.json.
error during build:
```
Error: Failed to resolve entry for package "@fluentui/react-icons-mdl2". The package may have incorrect main/module/exports specified in its package.json.

```

**Comment by rudymiked** [comment]
does anyone have a working solution?

**Comment by JustSlone** [comment]
Looks like there is a remaining issue with the react-icons-mdl2 package. 
Setting a resolution / pinning `@fluentui/react-icons-mdl2@1.3.37` should unblock you

**Comment by TristanWatanabe** [comment]
`@fluentui/react` version `8.109.1` is now out which should fix the issue
